tools/vsimporter/xib2nib/src/UIView.cpp

```diff
@@ -196,54 +209,82 @@ void UIView::InitFromStory(XIBObject* obj) {
+        if (strcmp(getAttrAndHandle("hidden"), "YES") == 0) {
             _hidden = true;
+        }
     }
     XIBObject* frameRect = FindMemberAndHandle("frame");
 
     if (frameRect) {
         _bounds.x = 0;
         _bounds.y = 0;
-        _bounds.width = strtod(frameRect->getAttrAndHandle("width"), NULL);
-        _bounds.height = strtod(frameRect->getAttrAndHandle("height"), NULL);
+        _bounds.width = static_cast<float>(strtod(frameRect->getAttrAndHandle("width"), NULL));
+        _bounds.height = static_cast<float>(strtod(frameRect->getAttrAndHandle("height"), NULL));
 
-        _center.x = strtod(frameRect->getAttrAndHandle("x"), NULL);
-        _center.y = strtod(frameRect->getAttrAndHandle("y"), NULL);
+        _center.x = static_cast<float>(strtod(frameRect->getAttrAndHandle("x"), NULL));
+        _center.y = static_cast<float>(strtod(frameRect->getAttrAndHandle("y"), NULL));
 
         _center.x += _bounds.width / 2.0f;
         _center.y += _bounds.height / 2.0f;
     }
 
     XIBObject* resizeMask = FindMemberAndHandle("autoresizingMask");
     if (resizeMask) {
-        if (resizeMask->getAttrAndHandle("widthSizable"))
+        if (resizeMask->getAttrAndHandle("widthSizable")) {
             _autoresizingMask |= (int)UIViewAutoresizingFlexibleWidth;
-        if (resizeMask->getAttrAndHandle("heightSizable"))
+        }
+
+        if (resizeMask->getAttrAndHandle("heightSizable")) {
             _autoresizingMask |= (int)UIViewAutoresizingFlexibleHeight;
-        if (resizeMask->getAttrAndHandle("flexibleMinX"))
+        }
+
+        if (resizeMask->getAttrAndHandle("flexibleMinX")) {
             _autoresizingMask |= (int)UIViewAutoresizingFlexibleLeftMargin;
-        if (resizeMask->getAttrAndHandle("flexibleMaxX"))
+        }
+
+        if (resizeMask->getAttrAndHandle("flexibleMaxX")) {
             _autoresizingMask |= (int)UIViewAutoresizingFlexibleRightMargin;
-        if (resizeMask->getAttrAndHandle("flexibleMinY"))
+        }
+
+        if (resizeMask->getAttrAndHandle("flexibleMinY")) {
             _autoresizingMask |= (int)UIViewAutoresizingFlexibleTopMargin;
-        if (resizeMask->getAttrAndHandle("flexibleMaxY"))
+        }
+
+        if (resizeMask->getAttrAndHandle("flexibleMaxY")) {
             _autoresizingMask |= (int)UIViewAutoresizingFlexibleBottomMargin;
+        }
     }
     _backgroundColor = (UIColor*)FindMemberAndHandle("backgroundColor");
 
     if (getAttrib("translatesAutoresizingMaskIntoConstraints")) {
-        if (strcmp(getAttrAndHandle("translatesAutoresizingMaskIntoConstraints"), "NO") == 0)
+        if (strcmp(getAttrAndHandle("translatesAutoresizingMaskIntoConstraints"), "NO") == 0) {
             _translatesAutoresizeToConstraints = false;
+        }
     }
 
-    _outputClassName = "UIView";
-}
+    if (getAttrib("horizontalHuggingPriority")) {
+        _horizontalHuggingPriority = strtof(getAttrAndHandle("horizontalHuggingPriority"), NULL);
```

Comment:
a cast isn't necessary here....?

---
In reply to: [107240874](https://github.com/Microsoft/WinObjC/pull/2269#discussion_r107240874) [](ancestors = 107240874)